File: app/src/main/java/education/mahmoud/quranyapp/data_layer/local/room/ReadLogDAO.java
package education.mahmoud.quranyapp.data_layer.local.room;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

@Dao
public interface ReadLogDAO {

    @Insert
    public void addReadLog(ReadLog item);

    @Update
    public void updateReadLog(ReadLog item);

    @Query("select * from readlog order by date desc")
    public List<ReadLog> getAllReadLog();

    @Query("select * from readlog where date = :date")
    ReadLog getReadLogBydate(long date);

    @Query("select * from readlog where strDate = :currentDate")
    ReadLog getReadLogBydate(String currentDate);

    @Delete
    void deleteReadLog(ReadLog readLog);
}

File: app/src/main/java/education/mahmoud/quranyapp/datalayer/local/room/ReadLogDAO.kt
package education.mahmoud.quranyapp.datalayer.local.room

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query
import androidx.room.Update

@Dao
interface ReadLogDAO {
    @Insert
    fun addReadLog(item: ReadLog)

    @Update
    fun updateReadLog(item: ReadLog)

    @get:Query("select * from readlog order by date desc")
    val allReadLog: List<ReadLog>

    @Query("select * from readlog where date = :date")
    fun getReadLogBydate(date: Long): ReadLog?

    @Query("select * from readlog where strDate = :currentDate")
    fun getReadLogBydate(currentDate: String?): ReadLog?

    @Delete
    fun deleteReadLog(readLog: ReadLog)
}